src/util/convertHTMLtoJSON.ts

```diff
@@ -0,0 +1,156 @@
+      }) as PreBlock)
+      return parentsWithChildren.length === 1 ? parentsWithChildren[0] : parentsWithChildren
+    }
+    if (nodes.every(node => Array.isArray(node))) return nodes.flat()
+    return nodes as PreBlock[]
+  }
+
+  /** Retrive content of Text element and return PreBlock. */
+  const convertText = (node: Text, wrappedTag?: string) => {
+    const content = node.content.includes('\n') ? node.content.trim() : node.content
+    return {
+      scope: wrappedTag ? `<${wrappedTag}>${content}</${wrappedTag}>` : content,
+      children: []
+    } as PreBlock
+  }
+
+  /** Convert to PreBlock based on foramtting tag. */
+  const convertFormattingTags = (node: Element) => {
+    if (node.tagName === 'i' || node.tagName === 'b') {
+      const [child] = node.children as Text[]
+      return { ...convertText(child, node.tagName), partOfThought: true }
+    }
+    else if (node.tagName === 'span') {
+      const attribute = getAttribute('class', node)
+      // WorkFlowy import with notes
+      if (attribute === 'note') {
+        const [note] = node.children
+        return {
+          scope: '=note',
+          children: [{
+            scope: note.type === 'text' ? note.content : '',
+            children: [],
+          }]
+        }
+      }
+      const [child] = node.children as Text[]
+      return { ...convertText(child), partOfThought: true }
+    }
+  }
+
+  /** Convert PreBlock array to Block array. */
+  const convertToBlock = (nodes: PreBlock[]): Block[] => nodes.map(node => {
+    if (node.children.length > 0) return { ...node, children: convertToBlock(node.children) } as Block
+    else return node as Block
+  })
+
+  /** Converts each Array of HimalayaNodes to PreBlock. */
+  const convert = (nodes: (Element | Text)[]): PreBlock[] | PreBlock => {
+    const blocks = nodes.map(node => {
+      // convert Text directly to PreBlock
+      if (node.type === 'text') {
+        return convertText(node)
+      }
+      // convert formatting tag
+      if (node.tagName === 'i' || node.tagName === 'b' || node.tagName === 'span') {
+        return convertFormattingTags(node)
+      }
+      // convert children of ul
+      if (node.tagName === 'ul') {
+        return convert(node.children as (Element | Text)[])
+      }
+      // convert li
+      const { children } = node
+      if (children.length === 1) {
+        const [childNode] = children as (Element | Text)[]
+        if (childNode.type === 'text') {
+          return {
+            scope: childNode.content,
+            children: []
+          } as PreBlock
+        }
+        else if (childNode.type === 'element' && childNode.tagName === 'ul') {
+          return {
+            scope: '',
+            children: convert(childNode.children as (Element | Text)[])
+          } as PreBlock
+        }
+      }
+      else return convert(node.children as (Element | Text)[])
+    }).filter(node => node !== undefined) as (PreBlock | PreBlock[])[]
```

Comment:
Let's use an explicit `return null` in the appropriate place within the mapping function instead of the implicit `undefined`. Functions should never return `undefined` in my opinion unless they are a void function (i.e. always return `undefined`).